<a href="https://colab.research.google.com/github/ValerySiccha/C8280/blob/main/CLI.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# SOLUCIÓN DE TAREA CLI. 



SICCHA HUAYANAY, VALERY KRYSTAL

**1) Mire en la parte superior e inferior de data.tsv entregado en esta actividad. Piensa en cómo se estructuran los datos. ¿Qué crees que significa cada columna? ¿Qué crees que significa cada fila?**

Rpta1: Cada columna significa los datos que irán ingresando y que se quedarán almacenados dentro de la base de datos .tsv; por otro lado, las filas son las carácterísticas, como son los ids que tendrán cada platillo, la cantidad pedida por platillo, los nombres de los platillos pedidos, las descripciones que tendrá el platillo (Los cuales pueden o no tener una (NULL or NaN)) y el precio que tienen estos.


*Mediante uso de líneas de comando:*

In [1]:
!head data.tsv

order_id	quantity	item_name	choice_description	item_price
1	1	Chips and Fresh Tomato Salsa	NULL	$2.39 
1	1	Izze	[Clementine]	$3.39 
1	1	Nantucket Nectar	[Apple]	$3.39 
1	1	Chips and Tomatillo-Green Chili Salsa	NULL	$2.39 
2	2	Chicken Bowl	[Tomatillo-Red Chili Salsa (Hot), [Black Beans, Rice, Cheese, Sour Cream]]	$16.98 
3	1	Chicken Bowl	[Fresh Tomato Salsa (Mild), [Rice, Cheese, Sour Cream, Guacamole, Lettuce]]	$10.98 
3	1	Side of Chips	NULL	$1.69 
4	1	Steak Burrito	[Tomatillo Red Chili Salsa, [Fajita Vegetables, Black Beans, Pinto Beans, Cheese, Sour Cream, Guacamole, Lettuce]]	$11.75 
4	1	Steak Soft Tacos	[Tomatillo Green Chili Salsa, [Pinto Beans, Cheese, Sour Cream, Lettuce]]	$9.25 


In [2]:
!tail data.tsv

1831	1	Chips	NULL	$2.15 
1831	1	Bottled Water	NULL	$1.50 
1832	1	Chicken Soft Tacos	[Fresh Tomato Salsa, [Rice, Cheese, Sour Cream]]	$8.75 
1832	1	Chips and Guacamole	NULL	$4.45 
1833	1	Steak Burrito	[Fresh Tomato Salsa, [Rice, Black Beans, Sour Cream, Cheese, Lettuce, Guacamole]]	$11.75 
1833	1	Steak Burrito	[Fresh Tomato Salsa, [Rice, Sour Cream, Cheese, Lettuce, Guacamole]]	$11.75 
1834	1	Chicken Salad Bowl	[Fresh Tomato Salsa, [Fajita Vegetables, Pinto Beans, Guacamole, Lettuce]]	$11.25 
1834	1	Chicken Salad Bowl	[Fresh Tomato Salsa, [Fajita Vegetables, Lettuce]]	$8.75 
1834	1	Chicken Salad Bowl	[Fresh Tomato Salsa, [Fajita Vegetables, Pinto Beans, Lettuce]]	$8.75 



**2) ¿Cuántos pedidos parece haber?**

Rpta2: Como se puede ver a simple vista y mediante el código de abajo, podemos notar que hay un total de 1834 pedidos que se realizaron.

*Mediante uso de pandas:*

In [ ]:
import pandas as pd

url='https://raw.githubusercontent.com/kapumota/Cuadernos/main/CLI/data.tsv'

df = pd.read_table(url)

valor= df['order_id'].value_counts()

print(valor)

926     23
1483    14
205     12
759     11
1786    11
        ..
768      1
341      1
1048     1
94       1
1199     1
Name: order_id, Length: 1834, dtype: int64


*Mediante uso de líneas de comando:*

In [43]:
#Podemos utilizar también el código anterior de !tail data.tsv 

!tail data.tsv

1831	1	Chips	NULL	$2.15 
1831	1	Bottled Water	NULL	$1.50 
1832	1	Chicken Soft Tacos	[Fresh Tomato Salsa, [Rice, Cheese, Sour Cream]]	$8.75 
1832	1	Chips and Guacamole	NULL	$4.45 
1833	1	Steak Burrito	[Fresh Tomato Salsa, [Rice, Black Beans, Sour Cream, Cheese, Lettuce, Guacamole]]	$11.75 
1833	1	Steak Burrito	[Fresh Tomato Salsa, [Rice, Sour Cream, Cheese, Lettuce, Guacamole]]	$11.75 
1834	1	Chicken Salad Bowl	[Fresh Tomato Salsa, [Fajita Vegetables, Pinto Beans, Guacamole, Lettuce]]	$11.25 
1834	1	Chicken Salad Bowl	[Fresh Tomato Salsa, [Fajita Vegetables, Lettuce]]	$8.75 
1834	1	Chicken Salad Bowl	[Fresh Tomato Salsa, [Fajita Vegetables, Pinto Beans, Lettuce]]	$8.75 



**3) ¿Cuántas líneas hay en este archivo?**

Rpta3: En data.tsv se puede observar, al final de todo el archivo, que hay 4623 líneas de código mientras que mediante el código hecho con pandas aparece que hay 4622 líneas de código. Por otro lado, utilizando líneas de comando se ve que son 4624 líneas de código

*Mediante uso de pandas:*

In [ ]:
df = pd.read_table(url)

print(df)

      order_id  quantity                              item_name  \
0            1         1           Chips and Fresh Tomato Salsa   
1            1         1                                   Izze   
2            1         1                       Nantucket Nectar   
3            1         1  Chips and Tomatillo-Green Chili Salsa   
4            2         2                           Chicken Bowl   
...        ...       ...                                    ...   
4617      1833         1                          Steak Burrito   
4618      1833         1                          Steak Burrito   
4619      1834         1                     Chicken Salad Bowl   
4620      1834         1                     Chicken Salad Bowl   
4621      1834         1                     Chicken Salad Bowl   

                                     choice_description item_price  
0                                                   NaN     $2.39   
1                                          [Clementine]  

*Mediante uso de líneas de comando:*

In [7]:
!cat data.tsv | wc -l


#cat es para imprimir los datos del archivo data.tsv mientras que wc -l es para contar el número de líneas

4624


**4) ¿Qué burrito es más popular, steak o chicken?**

Rpta4: Según el código de abajo tanto por pandas como por línea de comandos, el burrito más popular es el de ***pollo*** teniendo 553 pedidos mientras que el de carne solamente tiene 368.

*Mediante el uso de pandas:*

In [ ]:
sel = df.columns.isin(['item_name']) 
gua = df.columns[sel]
data= df[gua]

steak= data.groupby('item_name').get_group('Steak Burrito')
print(steak)

chicken= data.groupby('item_name').get_group('Chicken Burrito')
print(chicken)

          item_name
7     Steak Burrito
9     Steak Burrito
31    Steak Burrito
43    Steak Burrito
46    Steak Burrito
...             ...
4546  Steak Burrito
4607  Steak Burrito
4610  Steak Burrito
4617  Steak Burrito
4618  Steak Burrito

[368 rows x 1 columns]
            item_name
16    Chicken Burrito
17    Chicken Burrito
23    Chicken Burrito
29    Chicken Burrito
45    Chicken Burrito
...               ...
4556  Chicken Burrito
4558  Chicken Burrito
4559  Chicken Burrito
4566  Chicken Burrito
4569  Chicken Burrito

[553 rows x 1 columns]


*Mediante el uso de líneas de comandos:*

In [33]:
!grep -i 'steak burrito' data.tsv | wc -l

#grep es utilizado para buscar especificaciones utilizándose -i para que no distinga entre mayúsculas y minúsculas. 
#wc -l se explicó anteriomente

368


In [34]:
!grep -i 'chicken burrito' data.tsv | wc -l

553


**5) ¿Los burritos de pollo suelen tener frijoles negros o frijoles pintos?**

Rpta5: Según el código de abajo, los burritos suelen tener más *frijoles negros* que frijoles pintos.

In [44]:
!grep -i 'chicken burrito' data.tsv | grep -i 'black beans' | wc -l

#específicamos la búsqueda y sobre esa especificación volvemos a especificar otra búsqueda.

282


In [46]:
!grep -i 'chicken burrito' data.tsv | grep -i 'pinto beans' | wc -l

105


**6) Usa la línea de comandos para descubrir algo interesante sobre los datos dados.**

In [50]:
!head -c 1K data.tsv


#se recortó el data.tsv a los primeros 1000 carácteres.

order_id	quantity	item_name	choice_description	item_price
1	1	Chips and Fresh Tomato Salsa	NULL	$2.39 
1	1	Izze	[Clementine]	$3.39 
1	1	Nantucket Nectar	[Apple]	$3.39 
1	1	Chips and Tomatillo-Green Chili Salsa	NULL	$2.39 
2	2	Chicken Bowl	[Tomatillo-Red Chili Salsa (Hot), [Black Beans, Rice, Cheese, Sour Cream]]	$16.98 
3	1	Chicken Bowl	[Fresh Tomato Salsa (Mild), [Rice, Cheese, Sour Cream, Guacamole, Lettuce]]	$10.98 
3	1	Side of Chips	NULL	$1.69 
4	1	Steak Burrito	[Tomatillo Red Chili Salsa, [Fajita Vegetables, Black Beans, Pinto Beans, Cheese, Sour Cream, Guacamole, Lettuce]]	$11.75 
4	1	Steak Soft Tacos	[Tomatillo Green Chili Salsa, [Pinto Beans, Cheese, Sour Cream, Lettuce]]	$9.25 
5	1	Steak Burrito	[Fresh Tomato Salsa, [Rice, Black Beans, Pinto Beans, Cheese, Sour Cream, Lettuce]]	$9.25 
5	1	Chips and Guacamole	NULL	$4.45 
6	1	Chicken Crispy Tacos	[Roasted Chili Corn Salsa, [Fajita Vegetables, Rice, Black Beans, Cheese, Sour Cream]]	$8.75 
6	1	Chicken Soft Tacos	[Roasted Chili Co